### PR TITLE
finished awsResourcesDeploy class

### PR DESCRIPTION
**Pull request by eahefnawy**

### CODE REVIEW

File: lib/classes/CLI.js [hunk 1]
```diff
@@ -8,9 +8,9 @@ const os = require('os');
 const chalk = require('chalk');
 
 class CLI {
-  constructor(serverless, isInteractive, loadedPlugins, inputArray) {
+  constructor(serverless, interactive, loadedPlugins, inputArray) {
     this.serverless = serverless;
-    this.isInteractive = isInteractive;
+    this.interactive = interactive;
```
Comment (pmuens) [inline lib/classes/CLI.js]: Do we need this if we access it thourgh the `isInteractive()` method below?

Comment (eahefnawy) [inline lib/classes/CLI.js]: yep sometimes we want to explicitly set to interactive/non-interactive at the code level without changing the env vars (ie during tests) ... but actually the `isInteractive()` method should be aware of this probably, I'm gonna make that change.

And I had to rename it cause it was conflicting with the method name.

File: lib/plugins/awsResourcesDeploy/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Create Plugin
```
Comment (pmuens) [inline lib/plugins/awsResourcesDeploy/README.md]: Should be `AWS Resources Deploy` or something similar

File: lib/plugins/awsResourcesDeploy/awsResourcesDeploy.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    // TODO validate region is valid aws
+
+    const config = {
+      region: this.options.region,
+    };
+    this.CloudFormation = new AWS.CloudFormation(config);
+    BbPromise.promisifyAll(this.CloudFormation, { suffix: 'Promised' });
+
+
+    console.log('');
```
Comment (pmuens) [inline lib/plugins/awsResourcesDeploy/awsResourcesDeploy.js]: Is this intended?

File: lib/plugins/awsResourcesDeploy/awsResourcesDeploy.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    this.serverless.cli.log('Deploying Resources to AWS...');
+    // this.serverless.cli.spinner().start(); - weird bug!
```
Comment (pmuens) [inline lib/plugins/awsResourcesDeploy/awsResourcesDeploy.js]: Have you run `npm update`? I've removed the spinner and added it afterwards (was removed and re-added to the `package.json` during those commits).

File: lib/plugins/awsResourcesDeploy/package.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "serverless-aws-resources-deploy-plugin",
```
Comment (pmuens) [inline lib/plugins/awsResourcesDeploy/package.json]: There was a proposal to define a plugin naming convention such as `serverless-plugin-X`. If we go this way autocomplete on NPM (and search in general) for example would work way better.

File: lib/plugins/awsResourcesDeploy/tests/awsResourcesDeploy.js [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+
+    // it('should start spinner', () => {
+    //   const spinnerStub = sinon.stub(awsResourcesDeploy.serverless.cli, 'spinner');
+    //   return awsResourcesDeploy.validate().then(() => {
+    //     expect(spinnerStub.calledOnce).to.be.equal(true);
+    //     awsResourcesDeploy.serverless.cli.spinner.restore();
+    //   });
+    // });
+
+    it('should throw error if cfTemplate is missing', () => {
```
Comment (pmuens) [inline lib/plugins/awsResourcesDeploy/tests/awsResourcesDeploy.js]: `should throw error` --> `throw an error` (same applies for tests below)

File: tests/all.js [hunk 7]
```diff
@@ -3,6 +3,7 @@
 process.env.DEBUG = '*';
 require('./config');
 
+// Serverless Core
```
Comment (pmuens) [inline tests/all.js]: Awesome! Love the separation!

File: lib/plugins/awsResourcesDeploy/awsResourcesDeploy.js (hunk 3 above)
Comment (eahefnawy) [inline lib/plugins/awsResourcesDeploy/awsResourcesDeploy.js]: yep, to give white space before the greeting

**Pull request merged**